client/state/plans/actions.js

```diff
@@ -1,25 +1,38 @@
 /**
+ * External dependencies
+ */
+import findIndex from 'lodash/findIndex';
+
+/**
  * Internal dependencies
  */
 import wpcom from 'lib/wp';
-
 import {
 	PLANS_RECEIVE,
 	PLANS_REQUEST,
 	PLANS_REQUEST_SUCCESS,
 	PLANS_REQUEST_FAILURE
 } from '../action-types';
 
+import { PLAN_FREE, PLAN_PERSONAL } from 'lib/plans/constants';
+import personalPlan from 'lib/plans/personal-plan';
 /**
  * Action creator function: RECEIVE
  *
  * @param {Array} plans - WordPress.com plans list
  * @return {Object} action object
  */
 export const plansReceiveAction = plans => {
+	const freePlanIndex = findIndex( plans, plan => plan.product_slug === PLAN_FREE );
+	const hasPersonalPlan = plans.some( ( { product_slug } ) => PLAN_PERSONAL === product_slug );
```

Comment:
okay well I should have suggested this when I made my first comments, but there is also another useful **lodash** helper here too

```js
import matchesProperty from 'lodash/matchesProperty'

const freePlanIndex = findIndex( plans, matchesProperty( 'product_slug', PLAN_FREE );
const hasPersonalPlan = plans.some( matchesProperty( 'product_slug', PLAN_PERSONAL );
```